closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -310,7 +335,8 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         } else {
             openssl_version = (*OSSL_version)(0); /* get OPENSSL_VERSION */
             /* Ensure the OpenSSL version is "OpenSSL 1.0.x" */
-            if (0 != strncmp(openssl_version, OPENSSL_VERSION_1_0, strlen(OPENSSL_VERSION_1_0))) {
+            ossl_ver = ver_to_l(openssl_version);
+            if(!(ossl_ver >= OPENSSL_VERSION_1_0_0 && ossl_ver < OPENSSL_VERSION_1_1_0)) {
```

Comment:
Missing parentheses and space after `if`.

Reply:
Edited.

Reply:
Why did you add the redundant `ossl_ver < 0` part of the test?
```c
            if ((ossl_ver < 0) || !((OPENSSL_VERSION_1_0_0 <= ossl_ver) && (ossl_ver < OPENSSL_VERSION_1_1_0))) {
```
can be
```c
            if (!((OPENSSL_VERSION_1_0_0 <= ossl_ver) && (ossl_ver < OPENSSL_VERSION_1_1_0))) {
```

Reply:
Because we check the return value of `sscanf()`, if it doesnt match 3 values, it will return -1.

Reply:
No. The return value indicates how may patterns were matched; see https://cplusplus.com/reference/cstdio/sscanf/.

Reply:
Removed `(ossl_ver < 0)`.